extensions/typescript-language-features/src/languageFeatures/fileConfigurationManager.ts

```diff
@@ -215,12 +215,16 @@ export default class FileConfigurationManager extends Disposable {
 	private getAutoImportFileExcludePatternsPreference(config: vscode.WorkspaceConfiguration, workspaceFolder: vscode.Uri | undefined): string[] | undefined {
 		return workspaceFolder && config.get<string[]>('autoImportFileExcludePatterns')?.map(p => {
 			// Normalization rules: https://github.com/microsoft/TypeScript/pull/49578
-			const slashNormalized = p.replace(/\\/g, '/');
-			const isRelative = /^\.\.?($|\/)/.test(slashNormalized);
+			const isRelative = /^\.\.?($|[\/\\])/.test(p);
+			// In TypeScript < 5.3, the first path component cannot be a wildcard, so we need to prefix
+			// it with a path root (e.g. `/` or `c:\`)
+			const wildcardPrefix = this.client.apiVersion.gte(API.v540)
+				? ''
+				: path.parse(this.client.toTsFilePath(workspaceFolder)!).root;
```

Comment:
@mjbvz can you help confirm that this non-null assertion is safe here? I looked at the implementation and the only way it can return `undefined` is

```ts
if (fileSchemes.disabledSchemes.has(resource.scheme)) {
  return undefined;
}
```

I feel like we wouldn’t make it this far if that were the case for the workspace folder, but I’m not confident.